src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/FixedAssetsInvestmentsServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+import uk.gov.companieshouse.web.accounts.validation.helper.ServiceExceptionHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class FixedAssetsInvestmentsServiceImpl implements FixedAssetsInvestmentsService {
+
+    @Autowired
+    private FixedAssetsInvestmentsTransformer fixedAssetsInvestmentsTransformer;
+
+    @Autowired
+    private ApiClientService apiClientService;
+
+    @Autowired
+    private SmallFullService smallFullService;
+
+    @Autowired
+    private ValidationContext validationContext;
+
+    @Autowired
+    private ServiceExceptionHandler serviceExceptionHandler;
+
+    private static final UriTemplate FIXED_ASSETS_INVESTMENTS_URI =
+        new UriTemplate("/transactions/{transactionId}/company-accounts/{companyAccountsId}/small-full/notes/fixed-assets-investments");
+
+    private static final String INVALID_URI_MESSAGE = "Invalid URI for fixedAssetsInvestments resource";
+
+    private static final String RESOURCE_NAME = "fixed assets investments";
+
+    @Override
+    public FixedAssetsInvestments getFixedAssetsInvestments(String transactionId, String companyAccountsId, String companyNumber)
+        throws ServiceException {
+
+        FixedAssetsInvestmentsApi fixedAssetsInvestmentsApi = getFixedAssetsInvestmentsApi(transactionId, companyAccountsId);
+        FixedAssetsInvestments fixedAssetsInvestments = fixedAssetsInvestmentsTransformer.getFixedAssetsInvestments(fixedAssetsInvestmentsApi);
+
+        return fixedAssetsInvestments;
+    }
+
+    @Override
+    public List<ValidationError> submitFixedAssetsInvestments(String transactionId, String companyAccountsId,
+                                              FixedAssetsInvestments fixedAssetsInvestments, String companyNumber) throws ServiceException {
+
+        ApiClient apiClient = apiClientService.getApiClient();
+
+        String uri = FIXED_ASSETS_INVESTMENTS_URI.expand(transactionId, companyAccountsId).toString();
+
+        SmallFullApi smallFullApi = smallFullService.getSmallFullAccounts(apiClient, transactionId, companyAccountsId);
+
+        FixedAssetsInvestmentsApi fixedAssetsInvestmentsApi = fixedAssetsInvestmentsTransformer.getFixedAssetsInvestmentsApi(fixedAssetsInvestments);
+
+        boolean fixedAssetsInvestmentsResourceExists = hasFixedAssetsInvestments(smallFullApi.getLinks());
+
+        try {
+            if (!fixedAssetsInvestmentsResourceExists) {
+                apiClient.smallFull().fixedAssetsInvestments().create(uri, fixedAssetsInvestmentsApi).execute();
+            } else {
+                apiClient.smallFull().fixedAssetsInvestments().update(uri, fixedAssetsInvestmentsApi).execute();
+            }
+        } catch (URIValidationException e) {
+            throw new ServiceException(INVALID_URI_MESSAGE, e);
+        } catch (ApiErrorResponseException e) {
+            if (e.getStatusCode() == HttpStatus.BAD_REQUEST.value()) {
```

Comment:
Can we refactor this to use the new `ServiceExceptionHandler` class? It'll cut out duplication and resolve Sonar complaints. The same change can be applied to the employees service which has already been merged into `develop`.